server/xpub-server/entities/manuscript/resolvers/uploadManuscript.js

```diff
@@ -110,7 +123,7 @@ async function uploadManuscript(_, { file, id, fileSize }, { user }) {
     })
   }
 
-  manuscript.files.push(fileEntity)
+  addFileEntityToManuscript(manuscript, fileEntity)
```

Comment:
It looks like we save the file entity separately to begin with [here](https://github.com/elifesciences/elife-xpub/blob/5247cec4eacbce6bf18465c99114d4e25a1cc051/server/xpub-server/entities/manuscript/resolvers/uploadManuscript.js#L50-L55)

`manuscript.save` will also save the file entity to the database. This looks like it was existing behaviour, but if we're expecting there to be a single `MANUSCRIPT SOURCE` on a manuscript, we'll want to rework this logic.

Reply:
I would say we should expect the `file` to be deleted as it is now, however I know there is a concept of manuscript versioning in the future which could mean it would be preferable for the replaced `file` to continue to exist as it'll be linked to a different `manuscript`. I'll remove the initial `File.save()` for now then.

Reply:
`manuscript.save()` doesn't seem to automatically add the new `file` entity to the db, Errors get thrown when I remove the `.save()` when constructing the File object. Lets merge this in for now and I'll do a bit more digging into how `objection`'s save functionality works as I was also under the impression that it would create and destroy newly related entities on save.